src/Illuminate/Database/Query/Builder.php

```diff
@@ -899,14 +899,24 @@ public function whereNotNull($column, $boolean = 'and')
      * @param  string  $boolean
      * @param  bool  $not
      * @return $this
+     *
+     * @throws \InvalidArgumentException
      */
     public function whereBetween($column, array $values, $boolean = 'and', $not = false)
     {
+        if (count($values) !== 2) {
+            throw new InvalidArgumentException('Exactly 2 values required.');
```

Comment:
This message could probably be clearer, but I couldn't think of anything better.